pkg/controller/nodeipam/node_ipam_controller.go

```diff
@@ -133,7 +142,16 @@ func NewNodeIpamController(
 		case ipam.IPAMFromCloudAllocatorType:
 			cfg.Mode = nodesync.SyncFromCloud
 		}
-		ipamc, err := ipam.NewController(cfg, kubeClient, cloud, clusterCIDR, serviceCIDR, nodeCIDRMaskSize)
+
+		// we may end up here with no cidr at all in case of FromCloud/FromCluster
+		var cidr *net.IPNet
+		if len(clusterCIDRs) > 0 {
+			cidr = clusterCIDRs[0]
+		}
+		if len(clusterCIDRs) > 1 {
+			klog.Warningf("Multiple cidrs were configured with FromCluster or FromCloud. cidrs except first one were discarded")
```

Comment:
Not clear to me _why_ we're discarding them

Reply:
The `FromCluster` and `FromCloud` is the `why` part. Both are not configured for multiple cidrs.